### PR TITLE
Show filtered recipes

### LINKED ISSUE #11: User Story 7: Show filtered recipes
## Value statement 
As a **user** 
I want **to see filtered recipes** 
so that **I can choose one meal for all of my guests** 

## Description 
image

## Acceptance criteria
- [x] on the recipe page I can see correctly filtered recipes
- [x] there is a navbar to switch between home and recipe-page

## Tasks
- [x] create and add a component navbar (Home, Recipes) 
- [x] write a function to define the right URL for a GET-request based on the selected intolerances
- [x] fetch filtered data from the api
- [x] create RecipeCard-Component (recipe name, image, time)
- [x] render RecipeCards on the recipe-page

## Size
medium

### PR DESCRIPTION
**Pull request by s-kond**
fixes #11 

For now, you'll always see filtered pasta recipes. I'll add a searchbar in one of the next user stories.

### CODE REVIEW

File: src/pages/Recipes.js [hunk 1]
```diff
@@ -70,7 +87,12 @@ export default function Recipes() {
           <span key={item}> {item}</span>
         ))}
       </StyledSection>
-      <button onClick={() => navigate("/")}>back</button>
+      <section>
+        {data.map((recipe) => {
+          return <RecipeCard recipeData={recipe} />;
```
Comment (mpagels) [inline src/pages/Recipes.js]: I think you forgot the `key` prop.
And you could use the implicit return and get rid of the return statement.

File: src/pages/Recipes.js [hunk 2]
```diff
@@ -1,14 +1,30 @@
...
+
+  useEffect(() => {
+    fetchData();
+  }, []);
+
+  async function fetchData() {
+    const url = `https://api.edamam.com/api/recipes/v2?type=public&q=pasta&app_id=93fe14fb&app_key=515cbd62c9864645af6de1076420bdb9${intolerances
+      .map((item) => `&health=${item}`)
+      .join("")}`;
+    console.log(url);
```
Comment (mpagels) [inline src/pages/Recipes.js]: Here is a lonely `console.log`

File: src/pages/Recipes.js [hunk 3]
```diff
@@ -1,14 +1,30 @@
...
+  const [data, setData] = useState([]);
+
+  useEffect(() => {
+    fetchData();
+  }, []);
+
+  async function fetchData() {
+    const url = `https://api.edamam.com/api/recipes/v2?type=public&q=pasta&app_id=93fe14fb&app_key=515cbd62c9864645af6de1076420bdb9${intolerances
+      .map((item) => `&health=${item}`)
+      .join("")}`;
```
Comment (mpagels) [inline src/pages/Recipes.js]: This chaining is unfortunately very verbose. I don't understand what is happening here. Maybe you could break that step down with good variable names.

For example
```js
const healthParams = intolerances.map((item) => `&health=${item}`).join("")
const url = `https://api.edamam.com/api/recipes/v2?type=public&q=pasta&app_id=93fe14fb&app_key=515cbd62c9864645af6de1076420bdb9${healthParams }`;
```

File: src/pages/Recipes.js [hunk 4]
```diff
@@ -1,14 +1,30 @@
...
   const selectedGuests = guestArray.filter((guest) => guest.selected);
   let intolerances = getIntolerances();
+  const [data, setData] = useState([]);
+
+  useEffect(() => {
+    fetchData();
+  }, []);
+
+  async function fetchData() {
+    const url = `https://api.edamam.com/api/recipes/v2?type=public&q=pasta&app_id=93fe14fb&app_key=515cbd62c9864645af6de1076420bdb9${intolerances
```
Comment (mpagels) [inline src/pages/Recipes.js]: Is that your `API` key in there? I recommend putting that in an `.env` file.

File: src/pages/Recipes.js [hunk 5]
```diff
@@ -1,14 +1,30 @@
...
+      .map((item) => `&health=${item}`)
+      .join("")}`;
+    console.log(url);
+    const response = await fetch(url);
+    const data = await response.json();
+    console.log(data.hits);
```
Comment (mpagels) [inline src/pages/Recipes.js]: `console.log` here.

**Review by mpagels — APPROVED** [review]
Found some lonely console.logs. I marked them and some small improvement suggestion. Please have a look. Good job.

**Comment by s-kond** [comment]
Thanks for your review @mpagels - I implemented your suggestions! 

Regarding the API-key: I put it in a .env-file and that file in .gitignore. 
But if no one has access to the API-key except for me, no one can see the recipes they want to fetch, right? 
Won't that be problematic for quality assurance?

**Comment by mpagels** [comment]
You don't want the `API` key visible in your code base, since the project is open source. The App (Vercel) is getting the API key out from the `.env` when building the app. It will work when deployed (for quality assurance).

**Pull request merged**